website/blog/2023-05-01-evolving-data-engineer-craft.md

```diff
@@ -0,0 +1,194 @@
+slug: evolving-data-engineer-craft
+
+authors: [sung_chung, kira_furuichi]
+
+tags: [dbt product updates]
+hide_table_of_contents: false
+
+date: 2023-05-01
+is_featured: true
+---
+
+# Data Engineers + dbt 1.5: Evolving the Craft for Scale
+
+I, Sung, entered the data industry by chance in Fall 2014. I was using this thing called audit command language (ACL) to automate debits equal credits for accounting analytics (yes, it’s as tedious as it sounds). I remember working my butt off in a hotel room in Des Moines, Iowa where the most interesting thing there was a Panda Express. It was late in the AM. I’m thinking about 2 am. And I took a step back and thought to myself, “Why am I working so hard for something that I just don’t care about with tools that hurt more than help?”
+<!--truncate-->
+
+I did lots of soul searching and deduced I loved analytics, but not the job and subject matter. My next gig was in consulting where I bootstrapped my way into data engineering and had to learn the whole gamut below.
+
+| Tech skills | Place in tech stack | Why it mattered at the time |
+| --- | --- | --- |
+| Airflow | Orchestrator | The industry standard to run data pipelines |
+| SQL | Lingua franca of data transformation | My business logic codified (think: revenue by month) |
+| Python | Lingua franca of data engineering | It’s how you use Airflow |
+| Terraform | Get infra ready for airflow Kubernetes cluster | Infrastructure automation |
+| Google Cloud | Cloud | Big customer footprint |
+| Amazon Web Services | Cloud | Big customer footprint |
+| dbt | The T in ELT | The reason people finally test their data with SQL |
+| BigQuery | Cloud data warehouse | A lot of my clients used this |
+
+These are still great skills to learn and maintain even six years after I learned them in 2017. Armed with them, I finally saw the magic of the modern data stack and what problems it could solve. It took my questionable little pipelines back in 2014 and made them gleam with a new shine (and reliability). I felt like what the cool kids call a **[10x data engineer](https://knowyourmeme.com/memes/10x-engineer)**. 
+
+However, as my skills grew, so did the problems. Big Data soon turned into data swamps left and right across the clients I implemented data pipelines for using the above stack. The charm of using these great tools lost its shine, and my excitement gradually gave way to exhaustion. Not because they’re bad tools, but because the problem space of managing giant mounds of data required something data tools are still wrestling with today: scale and control. I kept looking for goldilocks projects to save/make money for companies. I wanted to build prestige in my career. But in practice, I was babysitting brittle data pipelines. To enable dozens of people was mind-numbing, much less hundreds of data analysts to all work elegantly together.
```

Comment:
```suggestion
However, as my skills grew, so did the problems. Big Data eventually turned into data swamps, the charm of using these great tools lost its shine, and my excitement gradually gave way to exhaustion. Not because they’re bad tools, but because the problem space of managing giant mounds of data required something data tools are still wrestling with today: scale and control. I kept looking for goldilocks projects to save/make money for companies. I wanted to build prestige in my career. But in practice, I was babysitting brittle data pipelines. To enable dozens of people was mind-numbing, much less hundreds of data analysts to all work elegantly together.
```